### DISCUSSION TITLE
Refresh doesn't maintain all_columns()?

### DISCUSSION THREAD

**Discussion by williamflaherty** [piccolo-orm/piccolo#799] [Q&A] [answered]
I noticed this today when updating an object. My code looks like this: 
```
        contract = (
            await Contract.objects(Contract.address_id).where(Contract.id == req.id).first()
        )
        await contract.address_id.update(req.address.dict()).where(
            Address.id == contract.address_id.id
        )
        await contract.refresh([Contract.address_id])
        address = await Address.objects().get(Address.id == contract.address_id)
```
however when you refresh all the columns, or just the address_id column as I have here, the newly polled object replaces the nested fk object with just the ID again. 

Is that right? I tried passing .all_columns() to the refresh call but to no avail.

**Comment by dantownsend** [comment, marked as answer]
Yeah, refresh doesn't refresh nested objects currently - it's something we should add though.

Try doing `await contract.address_id.refresh()`.
